src/ext/java/org/opentripplanner/ext/siri/AffectsMapper.java

```diff
@@ -203,6 +162,69 @@ private List<EntitySelector> mapVehicleJourneys(AffectsScopeStructure.VehicleJou
           );
         }
       }
+
+      final List<DatedVehicleJourneyRef> datedVehicleJourneyReves = affectedVehicleJourney.getDatedVehicleJourneyReves();
+      if (isNotEmpty(datedVehicleJourneyReves)) {
+        List<FeedScopedId> tripIds = new ArrayList<>();
+        for (DatedVehicleJourneyRef datedVehicleJourneyRef : datedVehicleJourneyReves) {
+          FeedScopedId tripIdFromDatedVehicleJourney = getTripId(
+            datedVehicleJourneyRef.getValue(),
```

Comment:
`getTripId` expects a SJ id. Now a DSJ id is passed to it. We would need to resolve the DSJ first, and use the service journey id from it in the EntitySelector.

Reply:
For some cornercases there might be an issue with requiring the TripId (either SJ or DSJ) to exist when creating these EntitySelectors. (This probably is already an issue today).

Usecase: A trip is added in realtime-data (SIRI ET, ExtraJourney), and a SIRI SX-message is also sent that affects the added trip.

Problem: If the SX-message is received/processed first, the `EntityResolver` will not find the trip, and the `AffectsMapper` will return an `EntitySelector.Unknown`. The SIRI SX-message will then not be applied correctly in the API until the SX-message is updated/processed again.

This may occur both when the actual trip is added, and when a new instance is started - and "a few" SX-messages is processed a lot faster than thousands of ET-messages during startup.

Would it possibly be better to add an `EntitySelector` for the Trip/StopAndTrip simply by calling `EntityResolver.resolveId(...)` regardless of which type of id (SJ/DSJ) is provided in the AffectsMapper - in effect just creating a simple `FeedScopedId`? 

Inclusion of these SIRI SX-messages would then be resolved based on the EntitySelector alone when requested.

Reply:
I'm not a Siri/Netex expert but I think this is sensible.

Reply:
The `AffectsMapper` now uses the `EntityResolver`, but is also more lenient - as in _not_ requiring trip to exist/be found when processing a SIRI SX-message.

Reply:
>For some cornercases there might be an issue with requiring the TripId (either SJ or DSJ) to exist when creating these EntitySelectors. (This probably is already an issue today).

Yes, this is and issue, and the solution is good. We might need to change the Affects GraphQL resolver to return an unknown entity, if we can not find the ServiceJourney when fetching the entities.